From e7d6861f51982bc0dcf5fc745c3d21c25bc2e604 Mon Sep 17 00:00:00 2001
From: Giuseppe Barillari <joe2k01dev@gmail.com>
Date: Fri, 29 Jun 2018 13:42:05 +0200
Subject: [PATCH 11/12] DisplaySettings: Allow devices to opt out from lift to
 wake detection

 * Some devices provide TYPE_WAKE_GESTURE sensor that isn't being used
   for lift to wake therefore we need to hide this feature on these devices.

Change-Id: Ia1b6737a02777dfcd3c80cab62f128a3090807fa
Signed-off-by: Giuseppe Barillari <joe2k01dev@gmail.com>
---
 res/values/config.xml                                       | 6 ++++++
 .../settings/display/LiftToWakePreferenceController.java    | 4 +++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index cc679a186f..46c5fc0685 100755
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -127,4 +127,10 @@
     <!-- Does the device allow for manual subscription provisioning? Only works for multi-sim devices,
          and currently depends on QC's proprietary telephony stack -->
     <bool name="config_enableManualSubProvisioning">true</bool>
+
+    <!-- Whether notify user that switch data service -->
+    <bool name="confirm_to_switch_data_service">false</bool>
+
+    <!-- Whether device supports lift to wake -->
+    <bool name="config_supportsLiftToWake">true</bool>
 </resources>
diff --git a/src/com/android/settings/display/LiftToWakePreferenceController.java b/src/com/android/settings/display/LiftToWakePreferenceController.java
index 7c2206b291..f888e36a8e 100644
--- a/src/com/android/settings/display/LiftToWakePreferenceController.java
+++ b/src/com/android/settings/display/LiftToWakePreferenceController.java
@@ -37,7 +37,9 @@ public class LiftToWakePreferenceController extends AbstractPreferenceController
     @Override
     public boolean isAvailable() {
         SensorManager sensors = (SensorManager) mContext.getSystemService(Context.SENSOR_SERVICE);
-        return sensors != null && sensors.getDefaultSensor(Sensor.TYPE_WAKE_GESTURE) != null;
+        return mContext.getResources().getBoolean(
+                com.android.settings.R.bool.config_supportsLiftToWake) &&
+                sensors != null && sensors.getDefaultSensor(Sensor.TYPE_WAKE_GESTURE) != null;
     }
 
     @Override
-- 
2.17.1

